packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,253 @@
+import React, { useCallback, SetStateAction, useRef } from 'react';
+import PropTypes from 'prop-types';
+import Portal from '@leafygreen-ui/portal';
+import { injectGlobal } from '@leafygreen-ui/emotion';
+import Icon, { Size } from '@leafygreen-ui/icon';
+import { useEventListener } from '@leafygreen-ui/hooks';
+import { uiColors } from '@leafygreen-ui/palette';
+import { css, cx } from '@leafygreen-ui/emotion';
+
+export const ModalSize = {
+  XXSmall: 'xxsmall',
+  XSmall: 'xsmall',
+  Small: 'small',
+  Default: 'default',
+  Large: 'large',
+  XLarge: 'xlarge',
+} as const;
+
+export type ModalSize = typeof ModalSize[keyof typeof ModalSize];
+
+const defaultSpacing = 18;
+
+const scrollContainer = css`
+  animation: fade-in 250ms ease-in-out;
+  -webkit-animation: fade-in 250ms ease-in-out;
+  background-color: ${uiColors.black};
+  opacity: 0.6;
+  overflow-y: auto;
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  display: flex;
```

Comment:
This combination of `display: flex`, `position: fixed`, and `overflow-y: auto` caused me issues in Firefox with modals that were long enough to require scrolling (I linked you the private repo PR in Slack)

Reply:
Fixed! Do you mind taking a look now?

Reply:
Hm. Soo.... my fix apparently broke a bunch of other stuff in the other repo. I'll let you know when I track down what happened.